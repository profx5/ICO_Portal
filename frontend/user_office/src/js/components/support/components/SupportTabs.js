import React from 'react';
import styled from 'styled-components';
import {Link} from 'react-router-dom';


const SupportTabs = ({ticketsAmount, activeTab, isTicketOpened, tabClickHandler}) => {
    return (
        <Wrapper>
            <Link to="/user_office/support"><Tab onClick={tabClickHandler.bind(this, 1)} active={!isTicketOpened && activeTab === 1 ? true : false}>New question</Tab></Link>
            <Link to="/user_office/support"><Tab onClick={tabClickHandler.bind(this, 2)} active={isTicketOpened || activeTab === 2 ? true : false}>My questions {ticketsAmount.length}</Tab></Link>
        </Wrapper>
    )
}


export default SupportTabs;

const Wrapper = styled.div`
    height: 56px;
`;

const Tab = styled.div`
    width: 220px;
    height: 100%;
    line-height: 56px;
    text-align: center;
    border-top-left-radius: 6px;
    border-top-right-radius: 6px;
    display: inline-block;
    color: ${props => props.active ? '#3172fd' : '#323c47'};
    background: ${props => props.active ? '#fff' : 'transparent'};
    cursor: pointer;
`;
