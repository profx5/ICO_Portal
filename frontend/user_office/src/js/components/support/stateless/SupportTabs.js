import React from 'react';
import styled from 'styled-components';
import {media} from 'js/utils/media';


const SupportTabs = ({ticketsAmount, activeTab, isTicketOpened, tabClickHandler}) => {
    return (
        <Wrapper>
            <Tab onClick={tabClickHandler.bind(this, 1)} active={!isTicketOpened && activeTab === 1 ? true : false}>New question</Tab>
            <Tab onClick={tabClickHandler.bind(this, 2)} active={isTicketOpened || activeTab === 2 ? true : false}>My questions {ticketsAmount.length}</Tab>
        </Wrapper>
    )
}


export default SupportTabs;

const Wrapper = styled.div`
    height: 56px;
    margin-top: 30px;
    ${media.xs} {
        display: flex;
        height: 45px;
    }
`;

const Tab = styled.div`
    width: 220px;
    height: 100%;
    line-height: 56px;
    text-align: center;
    border-top-left-radius: 6px;
    border-top-right-radius: 6px;
    display: inline-block;
    color: ${props => props.active ? '#3172fd' : '#323c47'};
    background: ${props => props.active ? '#fff' : 'transparent'};
    cursor: pointer;
    ${media.xs} {
        font-size: 14px;
        line-height: 45px;
        width: 50%;
        flex-basis: 50%;
    }
`;
