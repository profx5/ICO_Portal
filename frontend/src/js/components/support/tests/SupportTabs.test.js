import React from 'react';
import {configure, shallow} from 'enzyme';
import jest from 'jest-mock';
import Adapter from 'enzyme-adapter-react-16';

import SupportTabs from 'js/components/support/stateless/SupportTabs';


configure({ adapter: new Adapter() });

it('should render stateless SupportTabs', () => {
    const props = {
        tabClickHandler: jest.fn(),
        ticketsAmount: 0
    }
    const component = shallow(<SupportTabs {...props}/>);
    expect(component).toMatchSnapshot();
})
