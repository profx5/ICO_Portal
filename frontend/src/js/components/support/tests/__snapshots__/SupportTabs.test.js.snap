// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render stateless SupportTabs 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SupportTabs
    tabClickHandler={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          active={true}
          onClick={[Function]}
        >
          New question
        </ForwardRef>,
        <ForwardRef
          active={false}
          onClick={[Function]}
        >
          My questions 
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": true,
          "children": "New question",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "New question",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
    width: 220px;
    height: 100%;
    line-height: 56px;
    text-align: center;
    border-top-left-radius: 6px;
    border-top-right-radius: 6px;
    display: inline-block;
    color: ",
              [Function],
              ";
    background: ",
              [Function],
              ";
    cursor: pointer;
    ",
              "@media (max-width: 767px)",
              " {
        font-size: 14px;
        line-height: 45px;
        width: 50%;
        flex-basis: 50%;
    }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "active": false,
          "children": Array [
            "My questions ",
            undefined,
          ],
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "My questions ",
          undefined,
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              "
    width: 220px;
    height: 100%;
    line-height: 56px;
    text-align: center;
    border-top-left-radius: 6px;
    border-top-right-radius: 6px;
    display: inline-block;
    color: ",
              [Function],
              ";
    background: ",
              [Function],
              ";
    cursor: pointer;
    ",
              "@media (max-width: 767px)",
              " {
        font-size: 14px;
        line-height: 45px;
        width: 50%;
        flex-basis: 50%;
    }
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    height: 56px;
    margin-top: 30px;
    ",
          "@media (max-width: 767px)",
          " {
        display: flex;
        height: 45px;
    }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            active={true}
            onClick={[Function]}
          >
            New question
          </ForwardRef>,
          <ForwardRef
            active={false}
            onClick={[Function]}
          >
            My questions 
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": true,
            "children": "New question",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "New question",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": false,
              "rules": Array [
                "
    width: 220px;
    height: 100%;
    line-height: 56px;
    text-align: center;
    border-top-left-radius: 6px;
    border-top-right-radius: 6px;
    display: inline-block;
    color: ",
                [Function],
                ";
    background: ",
                [Function],
                ";
    cursor: pointer;
    ",
                "@media (max-width: 767px)",
                " {
        font-size: 14px;
        line-height: 45px;
        width: 50%;
        flex-basis: 50%;
    }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "active": false,
            "children": Array [
              "My questions ",
              undefined,
            ],
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            "My questions ",
            undefined,
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": false,
              "rules": Array [
                "
    width: 220px;
    height: 100%;
    line-height: 56px;
    text-align: center;
    border-top-left-radius: 6px;
    border-top-right-radius: 6px;
    display: inline-block;
    color: ",
                [Function],
                ";
    background: ",
                [Function],
                ";
    cursor: pointer;
    ",
                "@media (max-width: 767px)",
                " {
        font-size: 14px;
        line-height: 45px;
        width: 50%;
        flex-basis: 50%;
    }
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    height: 56px;
    margin-top: 30px;
    ",
            "@media (max-width: 767px)",
            " {
        display: flex;
        height: 45px;
    }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
